src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -371,8 +376,13 @@ private string InnerGenerateFile()
 					AnalyzerSuppressionsGenerator.Generate(writer, _analyzerSuppressions);
 
 					var controlBaseType = GetType(topLevelControl.Type);
+					var symbol = _generatorContext.Compilation.GetTypeByMetadataName($"{_className.ns}.{_className.className}");
+					if (symbol is null)
+                    {
+						throw new Exception($"Symbol '{_className.ns}.{_className.className}' was not found.");
+                    }
```

Comment:
`GetType` (like the one immediately above) is doing this validation and raising an exception. Is there something that you saw not working with it ?

Reply:
@jeromelaban I haven't noticed the method. Updated the code. Thanks!